app/models/article.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Article < ApplicationRecord
+  has_many :articles_sales, dependent: :restrict_with_error
+  has_many :sales, through: :articles_sales
+  has_many :articles_refunds, dependent: :restrict_with_error
+  has_many :refunds, through: :articles_refunds
+
+  validates :name, presence: true, allow_blank: false
+  validates :price, presence: true, allow_blank: false,
+                    numericality: { greater_than_or_equal_to: 0, only_integer: true, message: 'Must be a positive
```

Comment:
It's something that should be discuss. In a near or far future if we want to manage stocks of products maybe we'd like to sell things for free.

Reply:
If we are not sure it's something we need now, I would suggest allowing a price only strictly greater than 0. I think it would simplify some logic (not handling division by zero and a total price of zero). And that won't block us from adapting the logic in the future to support a price of 0